WordPress/Classes/ViewRelated/Post/PostListViewController.swift

```diff
@@ -0,0 +1,685 @@
+        
+        // If we have recently trashed posts, create an OR predicate to find posts matching the filter,
+        // or posts that were recently deleted.
+        if let recentlyTrashedPostObjectIDs = recentlyTrashedPostObjectIDs
+            where searchText?.characters.count == 0 && recentlyTrashedPostObjectIDs.count > 0 {
+            
+            let trashedPredicate = NSPredicate(format: "SELF IN %@", recentlyTrashedPostObjectIDs)
+            
+            if let originalFilterPredicate = filterPredicate {
+                filterPredicate = NSCompoundPredicate(orPredicateWithSubpredicates: [originalFilterPredicate, trashedPredicate])
+            } else {
+                filterPredicate = trashedPredicate
+            }
+        }
+        
+        if let filterPredicate = filterPredicate {
+            predicates.append(filterPredicate)
+        }
+        
+        if shouldShowOnlyMyPosts() {
+            let myAuthorID = blog?.account.userID ?? 0
+            
+            // Brand new local drafts have an authorID of 0.
+            let authorPredicate = NSPredicate(format: "authorID = %@ || authorID = 0", myAuthorID)
+            predicates.append(authorPredicate)
+        }
+        
+        if let searchText = searchText where searchText.characters.count > 0 {
+            let searchPredicate = NSPredicate(format: "postTitle CONTAINS[cd] %@", searchText)
+            predicates.append(searchPredicate)
+        }
+        
+        let predicate = NSCompoundPredicate(andPredicateWithSubpredicates: predicates)
+        return predicate
+    }
+    
+    // MARK: - Table View Handling
+    
+    private func tableView(tableView: UITableView, estimatedHeightForRowAtIndexPath indexPath: NSIndexPath) -> CGFloat {
+        let post = tableViewHandler?.resultsController.objectAtIndexPath(indexPath) as! Post
+        
+        if cellIdentifierForPost(post) == self.dynamicType.postCardRestoreCellIdentifier {
+            return CGFloat(self.dynamicType.postCardRestoreCellRowHeight)
+        }
+        
+        return CGFloat(self.dynamicType.postCardEstimatedRowHeight)
+    }
+    
+    private func tableView(tableView: UITableView, heightForRowAtIndexPath indexPath: NSIndexPath) -> CGFloat {
+        let width = CGRectGetWidth(tableView.bounds)
+        return self.tableView(tableView, heightForRowAtIndexPath: indexPath, forWidth: width)
+    }
+   
+    private func tableView(tableView: UITableView, heightForRowAtIndexPath indexPath: NSIndexPath, forWidth width: CGFloat) -> CGFloat {
+        guard let post = tableViewHandler?.resultsController.objectAtIndexPath(indexPath) as? Post else {
+            return 0
+        }
+        
+        if cellIdentifierForPost(post) == self.dynamicType.postCardRestoreCellIdentifier {
+            return CGFloat(self.dynamicType.postCardRestoreCellRowHeight)
+        }
+        
+        var cell : PostCardTableViewCell!
```

Comment:
> You mean initialize it to one, and change it afterwards if needed?

Not really.  Try this in a play ground:
```
var foo: String
foo = "foo"
foo

var bar: String!
bar

var baz: String
baz
```
`baz` is going to throw an exception because it is used before it is initialized, but both `foo` and `bar` will be fine. `foo` works because we initialize the variable (assign it a value) before it is used. `bar` works because we make it an explicitly unwrapped optional which means it gets a value of `nil` from the start. 

When I said `No need for the ! here. ` its because we're assigning a value before its used and there is never a situation were it can be used before it has a value.  Hence no need for the optional.